plugins/net_plugin/net_plugin.cpp

```diff
@@ -3252,18 +3285,6 @@ namespace eosio {
       // use c in this method instead of this to highlight that all methods called on c-> must be thread safe
       connection_ptr c = shared_from_this();
 
-      // if we have closed connection then stop processing
-      if( !c->socket_is_open() ) {
-         if( bsp ) {
-            // valid bsp means add_peer_block already called, need to remove it since we are not going to process the block
-            // call on dispatch strand to serialize with the add_peer_block calls
-            my_impl->dispatcher->strand.post( [blk_id]() {
-               my_impl->dispatcher->rm_block( blk_id );
-            } );
-         }
-         return;
-      }
```

Comment:
This very much looks like an anti-pattern to me. As the comment describes care has to be taken to call `rm_block` otherwise we may end up skipping a block. There is no reason we need to stop processing a block just because the socket has been closed since we received the block. It is cleaner to just to process the block as normal.

Reply:
No, the socket can be closed at any moment, so the code is already aware of that condition.